src/Symfony/Component/Validator/Constraints/TimezoneValidator.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * Validates whether a value is a valid timezone identifier.
+ *
+ * @author Javier Spagnoletti <phansys@gmail.com>
+ */
+class TimezoneValidator extends ConstraintValidator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$constraint instanceof Timezone) {
+            throw new UnexpectedTypeException($constraint, __NAMESPACE__.'\Timezone');
+        }
+
+        if (null === $value || '' === $value) {
+            return;
+        }
+
+        if (!is_scalar($value) && !(is_object($value) && method_exists($value, '__toString'))) {
+            throw new UnexpectedTypeException($value, 'string');
+        }
+
+        $value = (string) $value;
+        $zone = null !== $constraint->zone ? $constraint->zone : \DateTimeZone::ALL;
+
+        // @see: https://bugs.php.net/bug.php?id=75928
+        if ($constraint->countryCode) {
+            $timezoneIds = \DateTimeZone::listIdentifiers($zone, $constraint->countryCode);
+        } else {
+            $timezoneIds = \DateTimeZone::listIdentifiers($zone);
+        }
+
+        if ($timezoneIds && !in_array($value, $timezoneIds, true)) {
+            if ($constraint->countryCode) {
+                $code = Timezone::NO_SUCH_TIMEZONE_IN_COUNTRY_ERROR;
+            } elseif (null !== $constraint->zone) {
+                $code = Timezone::NO_SUCH_TIMEZONE_IN_ZONE_ERROR;
+            } else {
+                $code = Timezone::NO_SUCH_TIMEZONE_ERROR;
+            }
+
+            $violation = $this->context->buildViolation($constraint->message);
+
+            foreach ($this->generateValidationMessage($constraint->zone, $constraint->countryCode) as $placeholder => $message) {
+                $violation->setParameter($placeholder, $message);
+            }
+
+            $violation
+                ->setCode($code)
+                ->addViolation()
+            ;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefaultOption()
+    {
+        return 'zone';
+    }
+
+    /**
+     * Generates the replace parameters which are used in validation message.
+     *
+     * @param int|null    $zone
+     * @param string|null $countryCode
+     *
+     * @return array
+     */
+    private function generateValidationMessage(int $zone = null, string $countryCode = null): array
```

Comment:
i'd prefer not to have this method, the generated messages are also not translator friendly. Instead i would always set the (standard) `{{ value }}` parameter (being the zone ID in this case)